addons/pos_restaurant/static/src/js/models.js

```diff
@@ -212,27 +136,32 @@ patch(PosGlobalState.prototype, "pos_restaurant.PosGlobalState", {
     async _syncTableOrdersFromServer(tableId) {
         await this._removeOrdersFromServer(); // in case we were offline and we deleted orders in the mean time
         const ordersJsons = await this._getTableOrdersFromServer([tableId]);
+        await this._addPricelists(ordersJsons);
+        await this._addFiscalPositions(ordersJsons);
         const tableOrders = this.getTableOrders(tableId);
         this._replaceOrders(tableOrders, ordersJsons);
     },
-    async _syncAllOrdersFromServer() {
-        await this._removeOrdersFromServer(); // in case we were offline and we deleted orders in the mean time
-        const tableIds = [].concat(
-            ...this.floors.map((floor) => floor.tables.map((table) => table.id))
-        );
-        const ordersJsons = await this._getTableOrdersFromServer(tableIds); // get all orders
-        await this._syncTableOrdersToServer(); // to prevent losing the transferred orders
-        const allOrders = [...this.get_order_list()];
-        this._replaceOrders(allOrders, ordersJsons);
+    async _getOrdersJson() {
+        if (this.config.iface_floorplan) {
+            const tableIds = [].concat(
+                ...this.floors.map((floor) => floor.tables.map((table) => table.id))
+            );
+            await this._syncTableOrdersToServer(); // to prevent losing the transferred orders
+            const ordersJsons = await this._getTableOrdersFromServer(tableIds); // get all orders
+            return ordersJsons;
+        } else {
+            return await this._super();
+        }
     },
     _replaceOrders(ordersToReplace, newOrdersJsons) {
         ordersToReplace.forEach((order) => {
             // We don't remove the validated orders because we still want to see them in the ticket screen.
             // Orders in 'ReceiptScreen' or 'TipScreen' are validated orders.
-            if (order.server_id && !order.finalized && !this.transferredOrdersSet.has(order)) {
+            if ((!this.selectedOrder || (this.selectedOrder.uid != order.uid)) && order.server_id && !order.finalized && !this.transferredOrdersSet.has(order)) {
```

Comment:
Why add this?
And this is also broken, you've copied and put it in the point_of_sale module, so this is basically a complete override of the one in point_of_sale. So if you have both module installed, only this one will be run even in a normal point_of_sale